File: PreferenceSample/app/src/main/java/com/example/iyemon018/preferencesample/pref/GeneralPrefModel.java
package com.example.iyemon018.preferencesample.pref;

import com.os.operando.garum.annotations.Pref;
import com.os.operando.garum.annotations.PrefKey;
import com.os.operando.garum.models.PrefModel;

@Pref
public final class GeneralPrefModel extends PrefModel {
    
    @PrefKey("example_text")
    public String exampleText;
    
    @PrefKey("example_list")
    public String exampleList;
    
    @PrefKey("example_switch")
    public boolean exampleSwitch;
}

File: PreferenceSample/app/src/main/java/com/example/iyemon018/preferencesample/GeneralPrefModel.kt
package com.example.iyemon018.preferencesample

import com.os.operando.garum.annotations.Pref
import com.os.operando.garum.annotations.PrefKey
import com.os.operando.garum.models.PrefModel

@Pref()
class GeneralPrefModel : PrefModel() {

    private var initialized: Boolean = false

    @PrefKey("example_text")
    public var exampleText: String = ""
        set(value) {
            if (this.initialized) {
                field = value
            } else {
                if (value != "") {
                    field = value
                }
            }
        }

    @PrefKey("example_list")
    public var exampleList: String = ""

    @PrefKey("example_switch")
    public var exampleSwitch: Boolean = false

    init {
        this.initialized = true
    }

}